### PR TITLE
Develop humdrum

### PR DESCRIPTION
**Pull request by craigsapp**
Updates to humlib and iohumdrum.   This should also remove compiler warnings related to issue https://github.com/rism-digital/verovio/issues/3078

```
[  0%] Building CXX object CMakeFiles/verovio.dir/main.cpp.o
[  0%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/abbr.cpp.o
[  1%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/accid.cpp.o
[  1%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/add.cpp.o
[  2%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/anchoredtext.cpp.o
[  2%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/annot.cpp.o
[  3%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/app.cpp.o
[  3%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/areaposinterface.cpp.o
[  4%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/arpeg.cpp.o
[  4%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/artic.cpp.o
[  5%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/att.cpp.o
[  5%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/barline.cpp.o
[  6%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/bboxdevicecontext.cpp.o
[  6%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/beam.cpp.o
[  6%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/beamspan.cpp.o
[  7%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/beatrpt.cpp.o
[  7%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/boundingbox.cpp.o
[  8%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/bracketspan.cpp.o
[  8%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/breath.cpp.o
[  9%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/btrem.cpp.o
[  9%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/caesura.cpp.o
[ 10%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/choice.cpp.o
[ 10%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/chord.cpp.o
[ 11%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/clef.cpp.o
[ 11%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/controlelement.cpp.o
[ 12%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/corr.cpp.o
[ 12%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/course.cpp.o
[ 12%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/custos.cpp.o
[ 13%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/damage.cpp.o
[ 13%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/del.cpp.o
[ 14%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/devicecontext.cpp.o
[ 14%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/dir.cpp.o
[ 15%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/doc.cpp.o
[ 15%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/docselection.cpp.o
[ 16%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/dot.cpp.o
[ 16%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/drawinginterface.cpp.o
[ 17%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/durationinterface.cpp.o
[ 17%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/dynam.cpp.o
[ 18%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/editorial.cpp.o
[ 18%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/editortoolkit_cmn.cpp.o
[ 18%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/editortoolkit_neume.cpp.o
[ 19%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/elementpart.cpp.o
[ 19%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/ending.cpp.o
[ 20%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/expan.cpp.o
[ 20%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/expansion.cpp.o
[ 21%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/expansionmap.cpp.o
[ 21%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/f.cpp.o
[ 22%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/facsimile.cpp.o
[ 22%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/facsimileinterface.cpp.o
[ 23%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/fb.cpp.o
[ 23%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/featureextractor.cpp.o
[ 24%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/fermata.cpp.o
[ 24%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/fig.cpp.o
[ 25%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/fing.cpp.o
[ 25%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/floatingobject.cpp.o
[ 25%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/ftrem.cpp.o
[ 26%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/gliss.cpp.o
[ 26%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/glyph.cpp.o
[ 27%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/gracegrp.cpp.o
[ 27%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/graphic.cpp.o
[ 28%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/grpsym.cpp.o
[ 28%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/hairpin.cpp.o
[ 29%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/halfmrpt.cpp.o
[ 29%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/harm.cpp.o
[ 30%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/horizontalaligner.cpp.o
[ 30%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/instrdef.cpp.o
[ 31%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/io.cpp.o
[ 31%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/ioabc.cpp.o
[ 31%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/iodarms.cpp.o
[ 32%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/iohumdrum.cpp.o
[ 32%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/iomei.cpp.o
[ 33%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/iomusxml.cpp.o
[ 33%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/iopae.cpp.o
[ 34%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/keyaccid.cpp.o
[ 34%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/keysig.cpp.o
[ 35%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/label.cpp.o
[ 35%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/labelabbr.cpp.o
[ 36%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/layer.cpp.o
[ 36%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/layerdef.cpp.o
[ 37%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/layerelement.cpp.o
[ 37%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/lb.cpp.o
[ 37%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/lem.cpp.o
[ 38%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/ligature.cpp.o
[ 38%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/linkinginterface.cpp.o
[ 39%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/lv.cpp.o
[ 39%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/mdiv.cpp.o
[ 40%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/measure.cpp.o
[ 40%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/mensur.cpp.o
[ 41%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/metersig.cpp.o
[ 41%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/metersiggrp.cpp.o
[ 42%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/mnum.cpp.o
[ 42%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/mordent.cpp.o
[ 43%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/mrest.cpp.o
[ 43%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/mrpt.cpp.o
[ 43%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/mrpt2.cpp.o
[ 44%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/mspace.cpp.o
[ 44%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/multirest.cpp.o
[ 45%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/multirpt.cpp.o
[ 45%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/nc.cpp.o
[ 46%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/neume.cpp.o
[ 46%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/note.cpp.o
[ 47%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/num.cpp.o
[ 47%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/object.cpp.o
[ 48%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/octave.cpp.o
[ 48%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/options.cpp.o
[ 49%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/orig.cpp.o
[ 49%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/page.cpp.o
[ 50%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/pageelement.cpp.o
[ 50%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/pagemilestone.cpp.o
[ 50%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/pages.cpp.o
[ 51%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/pb.cpp.o
[ 51%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/pedal.cpp.o
[ 52%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/pgfoot.cpp.o
[ 52%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/pgfoot2.cpp.o
[ 53%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/pghead.cpp.o
[ 53%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/pghead2.cpp.o
[ 54%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/phrase.cpp.o
[ 54%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/pitchinflection.cpp.o
[ 55%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/pitchinterface.cpp.o
[ 55%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/plica.cpp.o
[ 56%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/plistinterface.cpp.o
[ 56%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/positioninterface.cpp.o
[ 56%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/proport.cpp.o
[ 57%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/rdg.cpp.o
[ 57%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/ref.cpp.o
[ 58%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/reg.cpp.o
[ 58%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/reh.cpp.o
[ 59%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/rend.cpp.o
[ 59%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/resources.cpp.o
[ 60%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/rest.cpp.o
[ 60%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/restore.cpp.o
[ 61%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/runningelement.cpp.o
[ 61%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/runtimeclock.cpp.o
[ 62%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/sb.cpp.o
[ 62%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/score.cpp.o
[ 62%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/scoredef.cpp.o
[ 63%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/scoredefinterface.cpp.o
[ 63%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/section.cpp.o
[ 64%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/sic.cpp.o
[ 64%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/slur.cpp.o
[ 65%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/space.cpp.o
[ 65%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/staff.cpp.o
[ 66%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/staffdef.cpp.o
[ 66%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/staffgrp.cpp.o
[ 67%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/stem.cpp.o
[ 67%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/subst.cpp.o
[ 68%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/supplied.cpp.o
[ 68%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/surface.cpp.o
[ 68%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/svg.cpp.o
[ 69%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/svgdevicecontext.cpp.o
[ 69%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/syl.cpp.o
[ 70%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/syllable.cpp.o
[ 70%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/symbol.cpp.o
[ 71%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/system.cpp.o
[ 71%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/systemelement.cpp.o
[ 72%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/systemmilestone.cpp.o
[ 72%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/tabdursym.cpp.o
[ 73%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/tabgrp.cpp.o
[ 73%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/tempo.cpp.o
[ 74%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/text.cpp.o
[ 74%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/textdirinterface.cpp.o
[ 75%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/textelement.cpp.o
[ 75%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/tie.cpp.o
[ 75%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/timeinterface.cpp.o
[ 76%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/timemap.cpp.o
[ 76%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/timestamp.cpp.o
[ 77%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/toolkit.cpp.o
[ 77%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/transposition.cpp.o
[ 78%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/trill.cpp.o
[ 78%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/tuning.cpp.o
[ 79%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/tuplet.cpp.o
[ 79%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/turn.cpp.o
[ 80%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/unclear.cpp.o
[ 80%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/verse.cpp.o
[ 81%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/verticalaligner.cpp.o
[ 81%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/view.cpp.o
[ 81%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/view_beam.cpp.o
[ 82%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/view_control.cpp.o
[ 82%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/view_element.cpp.o
[ 83%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/view_graph.cpp.o
[ 83%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/view_mensural.cpp.o
[ 84%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/view_neume.cpp.o
[ 84%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/view_page.cpp.o
[ 85%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/view_running.cpp.o
[ 85%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/view_slur.cpp.o
[ 86%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/view_tab.cpp.o
[ 86%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/view_text.cpp.o
[ 87%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/view_tuplet.cpp.o
[ 87%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/vrv.cpp.o
[ 87%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/zone.cpp.o
[ 88%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/hum/humlib.cpp.o
[ 88%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/crc/crc.cpp.o
[ 89%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/midi/Binasc.cpp.o
[ 89%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/midi/MidiEvent.cpp.o
[ 90%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/midi/MidiEventList.cpp.o
[ 90%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/midi/MidiFile.cpp.o
[ 91%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/midi/MidiMessage.cpp.o
[ 91%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/json/jsonxx.cc.o
[ 92%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/src/pugi/pugixml.cpp.o
[ 92%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/libmei/attconverter.cpp.o
[ 93%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/libmei/atts_analytical.cpp.o
[ 93%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/libmei/atts_cmn.cpp.o
[ 93%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/libmei/atts_cmnornaments.cpp.o
[ 94%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/libmei/atts_critapp.cpp.o
[ 94%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/libmei/atts_gestural.cpp.o
[ 95%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/libmei/atts_externalsymbols.cpp.o
[ 95%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/libmei/atts_facsimile.cpp.o
[ 96%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/libmei/atts_frettab.cpp.o
[ 96%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/libmei/atts_mei.cpp.o
[ 97%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/libmei/atts_mensural.cpp.o
[ 97%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/libmei/atts_midi.cpp.o
[ 98%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/libmei/atts_neumes.cpp.o
[ 98%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/libmei/atts_pagebased.cpp.o
[ 99%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/libmei/atts_shared.cpp.o
[ 99%] Building CXX object CMakeFiles/verovio.dir/Users/craig/git-cloud/verovio/libmei/atts_visual.cpp.o
[100%] Linking CXX executable verovio
[100%] Built target verovio
```

### CODE REVIEW

**Comment by craigsapp** [comment]
"All checks have passed", but is there any compiling done at this stage? (I could check for additional or new compiler warnings if so).

**Review by rettinghaus — APPROVED** [review]
Looks good for now. I'm sure I'll discover new things in the coming weeks. 🤓

**Comment by craigsapp** [comment]
I upgraded to MacOS 12.6 with clang version 14.0.0, and it did seem to generate new warnings that were not mentioned in MacOS 10.15.

**Comment by ahankinson** [comment]
@DavidBauer1984 could I ask you to please give this a review as well? @lpugin is on holiday, and my C++ isn't enough to catch the different standards / practices used in Verovio. @rettinghaus has already approved, so with two approvals I will merge it.

**Comment by craigsapp** [comment]
I have dictatorial control over the files `iohumdrum.cpp`, `iohumdrum.h`, `humlib.cpp`, and `humlib.h`  😉 .   In general it should be very safe for merging, since these files should only be able to causes crashes within verovio if someone imports Humdrum files. (And I only merge with `develop` about once every three months to give time to iron out any odd behaviors with changes in the code, so there are too many changes to review carefully.)

In any case this PR can also wait a week or so (but the warnings would be displaying until then).

**Comment by ahankinson** [comment]
Even dictators can get something wrong. I don't think it's a bad idea to have another pair of eyes on the code.

**Comment by craigsapp** [comment]
Dictators always do things right until they don't. 🇺🇦

File: include/hum/humlib.h
Comment (DavidBauer1984) [inline include/hum/humlib.h:5941]: (Code style) Getters should be `const`.

File: src/hum/humlib.cpp
Comment (DavidBauer1984) [inline src/hum/humlib.cpp:1093]: (Code style) std algorithms should be preferred instead of raw loops. In this case `std::accumulate` seems natural.

**Review by DavidBauer1984 — APPROVED** [review]
- Building `Verovio-Humdrum` in Xcode fails, but this is not your fault. I will open a PR today and fix this.
- The following example crashes: https://kern.humdrum.org/cgi-bin/ksdata?l=users/craig/classical/beethoven/piano/sonata&file=sonata01-1.krn&f=kern
I think, this is unrelated to the changes in the PR. The problem is that when creating editorial elements like `Choice`, they need to get a proper editorial level. Compare
```cpp
Choice *choice = new Choice();
```
in `iohumdrum.cpp` with
```cpp
Choice *vrvChoice = new Choice(level);
```
in `iomei`.
- I've added a couple of comments suggesting improvements for code style. Please have a look, it's not mandatory.

**Comment by ahankinson** [comment]
Thank you for your helpful comments @DavidBauer1984 ! These look sensible to me, but we'll see if @craigsapp is amenable to changing it.

**Pull request merged**
